database/option.go

```diff
@@ -148,45 +152,59 @@ func (f *OptionFactory) WithClusterOption() (app.Option, error) {
 }
 
 // ensureBindAddress sets the bind address, used by clients and peers.
-// TODO (manadart 2022-09-30): For now, this is *similar* to the peergrouper
-// logic in that we require a unique local-cloud IP. We will need to revisit
-// this because at present it is not influenced by a configured juju-ha-space.
+// We will need to revisit this because at present it is not influenced
+// by a configured juju-ha-space.
 func (f *OptionFactory) ensureBindAddress() error {
 	if f.bindAddress != "" {
 		return nil
 	}
 
-	sysAddrs, err := f.interfaceAddrs()
+	nics, err := net.Interfaces()
 	if err != nil {
-		return errors.Annotate(err, "querying addresses of system NICs")
+		return errors.Annotate(err, "querying local network interfaces")
 	}
 
-	var addrs network.MachineAddresses
-	for _, sysAddr := range sysAddrs {
-		var host string
+	var addrs corenetwork.MachineAddresses
+	for _, nic := range nics {
+		if ignoreInterface(nic) {
+			continue
+		}
 
-		switch v := sysAddr.(type) {
-		case *net.IPNet:
-			host = v.IP.String()
-		case *net.IPAddr:
-			host = v.IP.String()
-		default:
+		sysAddrs, err := nic.Addrs()
+		if err != nil || len(sysAddrs) == 0 {
 			continue
 		}
 
-		addrs = append(addrs, network.NewMachineAddress(host))
+		for _, addr := range sysAddrs {
+			switch v := addr.(type) {
+			case *net.IPNet:
+				addrs = append(addrs, corenetwork.NewMachineAddress(v.IP.String()))
+			case *net.IPAddr:
+				addrs = append(addrs, corenetwork.NewMachineAddress(v.IP.String()))
+			default:
+			}
+		}
 	}
 
-	cloudLocal := addrs.AllMatchingScope(network.ScopeMatchCloudLocal)
+	cloudLocal := addrs.AllMatchingScope(corenetwork.ScopeMatchCloudLocal).Values()
+
 	if len(cloudLocal) == 0 {
 		return errors.NotFoundf("suitable local address for advertising to Dqlite peers")
 	}
-	if len(cloudLocal) > 1 {
-		return errors.Errorf(
-			"multiple local-cloud addresses detected. Dqlite bootstrap requires a unique address; found %v", cloudLocal)
-	}
 
-	f.bindAddress = cloudLocal.Values()[0]
-	f.logger.Debugf("determined Dqlite bind address: %s", f.bindAddress)
+	sort.Strings(cloudLocal)
```

Comment:
Does the order actually matter and are we just trying to be deterministic here?

Reply:
Deterministic. We want the same one post-bootstrap when the worker fires up.